packages/components/src/navigation/menu-item.js

```diff
@@ -41,22 +41,36 @@ const NavigationMenuItem = ( props ) => {
 		onClick( props );
 	};
 
-	const LinkTagName = linkTag || Button;
+	const linkContents = (
+		<>
+			<Text variant="body.small">
+				<span>{ title }</span>
+			</Text>
+			{ badge && <BadgeUI>{ badge }</BadgeUI> }
+			{ hasChildren ? <Icon icon={ chevronRight } /> : null }
+		</>
+	);
 
 	return (
 		<MenuItemUI className={ classes }>
-			<LinkTagName
-				className={ classes }
-				href={ href }
-				onClick={ handleClick }
-				{ ...linkTagProps }
-			>
-				<Text variant="body.small">
-					<span>{ title }</span>
-				</Text>
-				{ badge && <BadgeUI>{ badge }</BadgeUI> }
-				{ hasChildren ? <Icon icon={ chevronRight } /> : null }
-			</LinkTagName>
+			{ LinkComponent ? (
+				<LinkComponent
+					className={ classes }
+					onClick={ handleClick }
```

Comment:
Missing `href` in here.  I think the old implementation simplifies the props logic here a bit.